 URL : "https://github.com/Hironsan/keras-crf-layer/issues/1" TITLE : error at the sample code. BODY : hi, i am not sure if this is an active repository for the public, but i would like to learn from your code to use crf in keras. i tried to learn your sample code in the readme.md and i receive below message. though i know crf, i am new to keras so it would be great if you have an idea of how to fix this. --------------------------------------------------------------------------- valueerror traceback most recent call last <ipython-input-5-8b3f4294bf42> in <module> ----> 1 pred = crf inputs=word_embeddings, sequence_lengths=sequence_lengths 2 model = model inputs= word_ids, sequence_lengths , outputs= pred 3 model.compile loss=crf.loss, optimizer='sgd' 4 5 train first 1 batch. /home/jbkoh/anaconda3/lib/python3.6/site-packages/keras-2.0.8-py3.6.egg/keras/engine/topology.py in __call__ self, inputs, kwargs 556 raise exceptions in case the input is not compatible 557 with the input_spec specified in the layer constructor. --> 558 self.assert_input_compatibility inputs 559 560 collect input shapes to build layer. /home/jbkoh/anaconda3/lib/python3.6/site-packages/keras-2.0.8-py3.6.egg/keras/engine/topology.py in assert_input_compatibility self, inputs 443 'but it received ' + str len inputs + 444 ' input tensors. input received: ' + --> 445 str inputs 446 for input_index, x, spec in enumerate zip inputs, input_spec : 447 if spec is none: valueerror: layer crf_layer_2 expects 2 inputs, but it received 1 input tensors. input received: <tf.tensor 'embedding_2/gather:0' shape= 2, 2, 11 dtype=float32> thanks for sharing your code!